app/models/report.rb

```diff
@@ -2,6 +2,9 @@ class Report < ActiveRecord::Base
   ALPHABET = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ"
   BASE = ALPHABET.length
 
+  validates :report, length: { minimum: 100, maximum: 20_000 }
```

Comment:
Maybe these numbers can be moved into constants? That way we could get the value externally like `Report::MAX_LENGTH` or something like that in the tests for example.

Reply:
I don't love the idea of extracting the constant only to be used externally (like from within a test) -- I prefer to leave the values in this line because it is self documented. I think adding constants doesn't add much value here. Maybe in the future we will need them, but for now I'll leave this as is.